src/toolchain/export-web.md

```diff
@@ -0,0 +1,155 @@
+
+
+# Intro to Web Builds
+
+Web builds are a fair bit more difficult to get started with compared to native builds.
+This will be a complete guide on how to get things compiled.
+However, setting up a web server to host and share your game is considered out of scope of this guide, and is best explained elsewhere.
+
+Information to consider before starting:
+
+```admonish warning
+Web support with `gdext` is experimental and should be understood as such before proceeding.
+```
+
+---
+
+
+## Installing Pre-Requisites
+
+Install a nightly build of `rustc`, the `wasm32-unknown-emscripten` target for `rustc`, and `rust-src`.
+The reason that a nightly `rustc` is required is that nightly is required to build `std` (`-Zbuild-std`).
+Assuming that Rust was installed with `rustup`, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+Next, install Emscripten.  The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+```sh
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install 3.1.39
+./emsdk activate 3.1.39
+source ./emsdk.sh     (or ./emsdk.bat on windows)
+```
+
+It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+This is platform-specific.
+
+---
+
+
+## Project Configuration
+
+Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
+
+```toml
+godot = { git = "https://github.com/godot-rust/gdext", branch = "master", features = ["experimental-wasm", "lazy-function-tables"] }
+```
+
+If you do not already have `.cargo/config.toml` file, do the following:
+
+- Create a `.cargo` directory at the same level as your `Cargo.toml`.
+- Inside of that directory, create a `config.toml` file.
+
+The contents of this file need to contain the following:
+
+```toml
+[target.wasm32-unknown-emscripten]
+rustflags = [
+    "-C", "link-args=-sSIDE_MODULE=2",
+    "-C", "link-args=-sUSE_PTHREADS=1",
+    "-C", "target-feature=+atomics,+bulk-memory,+mutable-globals",
+    "-Zlink-native-libraries=no"
+]
+```
+
+Edit the project's `.gdextension` file to include support for web exports.
+This file will probably be at `godot/{YourCrate}.gdextension`.
+The format will be similar to the following:
+
+```gdextension
+web.debug.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/debug/EXTENSIONNAME.wasm"
+web.release.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/release/EXTENSIONNAME.wasm"
```

Comment:
```suggestion
...

[libraries]
...
web.debug.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/debug/EXTENSIONNAME.wasm"
web.release.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/release/EXTENSIONNAME.wasm"
```

<br>

Also, would you need to replace `EXTENSIONNAME` with `{YourCrate}` here?

Reply:
Not sure about the flag! Actually didn't notice that I had an older one on my personal config file.